File: platform/sdk/nodejs/examples/cloud/cloud-022_worm_retention_1y.js
const { CloudClient } = require("../../mox-sdk-cloud");
const client = new CloudClient({ region: "cn-east-1" });
client.createBucket("worm-bucket");
client.putObject("worm-bucket", "critical.txt", "important data");
const result = client.wormRetention1y("worm-bucket", "critical.txt");
if (!result.ok || result.retention.days !== 365) process.exit(1);
console.log("XJ-OK: cloud-022_worm_retention_1y");
process.exit(0);

File: platform/sdk/python/examples/cloud/cloud-022_worm_retention_1y.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from mox_sdk_cloud import CloudClient

client = CloudClient({"region": "cn-east-1"})
client.createBucket("worm-bucket")
client.putObject("worm-bucket", "critical.txt", "important data")
result = client.wormRetention1y("worm-bucket", "critical.txt")
if not result["ok"] or result["retention"]["days"] != 365:
    sys.exit(1)
print("XJ-OK: cloud-022_worm_retention_1y")
sys.exit(0)
